### ISSUE TITLE
Pmml permitted parameters

### ISSUE THREAD

**Issue by rahuls04** [jpmml/jpmml-evaluator#202]
Am I able to pass a list as an argument to a pmml file?

**Comment by vruusmann** [comment]
> Am I able to pass a list as an argument to a pmml file?

This question is unclear to me.

Please provide some more context, example code etc.

**Comment by rahuls04** [comment]
I have a boolean that is an input to my logistic regression model. In production, I do not have access to this boolean. I need to engineer this boolean from a list, by checking if a certain value is present in the list. To do so, I would need to have a list as one of my <DataField> in the pmml, after which I would check for the value in Local Transformations and construct the boolean within the pmml. Just want to know if this is possible to do. Would something like 
`<DataField name="my_list" optype="continuous" dataType="list"/>  work?

**Comment by vruusmann** [comment]
> I need to engineer this boolean from a list, by checking if a certain value is present in the list.

What is statically known? The list, or the certain value?Or are they both variables?

Checking if a certain variable value is contained in a statically known list:
```xml
<DerivedField name="value_in_list" dataType="boolean" optype="categorical">
  <Apply function="isIn">
    <FieldRef field="my_value"/>
    <Constant>first list element</Constant>
    <Constant>second list element</Constant>
    <Constant>third list element</Constant>
  </Apply>
</DerivedField>
```

Turning the above around - sending a list to a PMML, and asking it to iterate over its elements and report back if one of them is equal to a specific value - is not recommended/possible.

The main reason is that PMML only defines scalar data types; there is not a single collection-type data type (list, set, collection) available right now. I've asked DMG.org to consider adding a collection data type (http://mantis.dmg.org/view.php?id=223), but in a typical DMG.org fashion, this won't be discussed until 2021 at earliest.

TLDR: Not possible in canonical PMML. But you can always write a custom Java-backed function to do whatever you want to do.
